File: Tercer Parcial/Proyectos/Galarza_Zurita/Lenguajes/ArbolBinario.py
class Arbol:

    def __init__(self, arbol, id, palabra):
        self.derecha = None
        self.izquierda = None
        self.id = id
        self.palabra = palabra

    def agregar(self, arbol, id, palabra):
        if arbol.id >= id:
            self.agregarIzquierda(arbol, id, palabra)
        elif arbol.id < id:
            self.agregarDerecha(arbol, id, palabra)

    def agregarIzquierda(self, arbol, id, palabra):
        if arbol.izquierda == None:
            arbol.izquierda = Arbol(arbol, id, palabra)
        else:
            self.agregar(arbol.izquierda, id, palabra)

    def agregarDerecha(self, arbol, id, palabra):
        if arbol.derecha == None:
            arbol.derecha = Arbol(arbol, id, palabra)
        else:
            self.agregar(arbol, id, palabra)

    def busquedaPalabra(self, arbol, id):
        while arbol != None:
            if arbol.id == id:
                return arbol.palabra
            arbol = arbol.derecha
        return "No disponible"

    def busquedaCodigo(self, arbol, palabra):
        while arbol != None:
            if arbol.palabra == palabra:
                return arbol.id
            arbol = arbol.derecha
        return 0

    def traduccion(self, tree, arbol, palabra):
        return self.busquedaPalabra(arbol, self.busquedaCodigo(tree, palabra))


tree = Arbol(None, 1, "shield")
arbol = Arbol(None, 1, "escudo")
pronun = Arbol(None, 1, "shild")
tree.agregar(tree,2,"house")
arbol.agregar(arbol,2,"casa")
pronun.agregar(pronun,2,"jaus")

print(tree.traduccion(tree, arbol, "shield"))
print(tree.traduccion(arbol,tree,"casa"))
print(tree.traduccion(arbol,pronun,"casa"))

File: Tercer Parcial/Proyectos/Galarza_Zurita/Lenguajes/ArbolBinario.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package diccionario;

/**
 *
 * @author Kevin Zurita
 */
public class ArbolBinario {
    
    Nodo raiz;
    public ArbolBinario(){
        raiz=null;
    }
    
    public void insertar(int id, String palabra){
        Nodo n = new Nodo(id,palabra);
        
        if(raiz==null){
            raiz=n;
        }else{
            Nodo aux=raiz;
            while(aux!=null){
                n.padre=aux;
                if(n.id>=aux.id){
                    aux=aux.derecha;
                }else{
                    aux=aux.izquierda;
                }
            }
            
            if(n.id<n.padre.id){
                n.padre.izquierda=n;
            }else{
                n.padre.derecha=n;
            }
        }
    }
    
    public String busquedaPalabra(Nodo n, int x){
        while (n != null)
	{
		if (n.id == x)
		{
			return n.palabra;
		}
		n = n.derecha;
	}
	return "No disponible.";
    }
    
    public int busquedaCodigo(Nodo n, String word){
        while (n != null)
	{
		if (n.palabra.equals(word))
		{
			return n.id;
		}
		n = n.derecha;
	}
        return 0;
    }
    
    public String traduccion(Nodo a, Nodo b, String word){
        return busquedaPalabra(b,busquedaCodigo(a,word));
    }
    
    private class Nodo{
        public Nodo padre;
        public Nodo izquierda;
        public Nodo derecha;
        public int id;
        public String palabra; 
        
        public Nodo(int key,String word){
            id=key;
            palabra=word;
            padre=null;
            izquierda=null;
            derecha=null;
        }
    }
}
